core/src/main/java/org/apache/iceberg/PartitionStatsUtil.java

```diff
@@ -53,14 +56,43 @@ public static Collection<PartitionStats> computeStats(Table table, Snapshot snap
 
     StructType partitionType = Partitioning.partitionType(table);
     List<ManifestFile> manifests = snapshot.allManifests(table.io());
-    Queue<PartitionMap<PartitionStats>> statsByManifest = Queues.newConcurrentLinkedQueue();
-    Tasks.foreach(manifests)
-        .stopOnFailure()
-        .throwFailureWhenFinished()
-        .executeWith(ThreadPools.getWorkerPool())
-        .run(manifest -> statsByManifest.add(collectStats(table, manifest, partitionType)));
+    return collectStats(table, manifests, partitionType).values();
+  }
+
+  /**
+   * Computes the partition stats incrementally after the given snapshot to current snapshot.
+   *
+   * @param table the table for which partition stats to be computed.
+   * @param afterSnapshot the snapshot after which partition stats is computed (exclusive).
+   * @param currentSnapshot the snapshot till which partition stats is computed (inclusive).
+   * @return the {@link PartitionMap} of {@link PartitionStats}
+   */
+  public static PartitionMap<PartitionStats> computeStatsIncremental(
```

Comment:
why do we need both? can't we always try to do incremental if prev snapshot stats file exists?

Reply:
I think it is good to have a way to force recompute stats, in case somehow the previous version of the stats become corrupted.

WDYT?

Reply:
Yes. I kept the full computation to force recompute if the stats are corrupted.

Reply:
agree, ````ALTER TABLE t COMPUTE STATS```` could be exercised for that. 
However, why not add just an extra param like 'forceRecompute' instead of diff method signatures?

Reply:
> However, why not add just an extra param like forceRecompute/recompute instead of diff method signatures?

Its a public interface added before 1.8.0. So, modifying it will cause API breakage. Hence, new interface. 
But they both reuse the same code once the manifest list is identified.

Reply:
we can overload a method and add an extra param, the previous one would use force flag

Reply:
I am checking how to simplify now.

Reply:
I was thinking about something like this, and since we don't have dependency on `data` here, we could do this in `PartitionStatsHandler` 
````
public static Collection<PartitionStats> computeStats(Table table, Snapshot snapshot) {
  return computeStats(table, snapshot, true);
}
public static Collection<PartitionStats> computeStats(Table table, Snapshot snapshot, boolean recompute) {
 ....
  if (!recompute && latestStatsFile != null) {
    load incremental
  }
}
````

Reply:
Not recommended to use boolean argument for public interface

https://iceberg.apache.org/contribute/?h=contribute#boolean-arguments

I have some refactoring in mind. I will finish in an hour and push it.

Reply:
few lines lower
````
  // GOOD: these uses of dropTable are clear to the reader
  dropTable(identifier, true /* purge data */);
  dropTable(identifier, purge);
````

Reply:
WDYM it's not public - that is the Catalog API

Reply:
I was looking for a public keyword :) 

But yeah, better to follow the guidelines as mentioned above. It is more cleaner. 
I still refactored this to accept `boolean` in a private method of `PartitionStatsHandler`